### DISCUSSION TITLE
3.2.9 install on Windows

### DISCUSSION THREAD

**Discussion by JayTea144** [PodcastGenerator/PodcastGenerator#737] [Q&A] [answered]
Hi all,

I was planning on testing this (PodcastGenerator 3.2.9) on my Windows device with Apache/2.4.54 and PHP Version 8.2.0.

I get to Step 3 asking for credentials and when I hit submit I get the error below.

> Deprecated: Optional parameter $path declared before required parameter $key is implicitly treated as a required parameter in C:\Apache24\htdocs\podcast\core\misc\configsystem.php on line 123

>Warning: require_once(C:Apache24htdocspodcast/vendor/james-heinrich/getid3/getid3/getid3.php): Failed to open stream: No such file or directory in C:\Apache24\htdocs\podcast\core\episodes.php on line 211

>Fatal error: Uncaught Error: Failed opening required 'C:Apache24htdocspodcast/vendor/james-heinrich/getid3/getid3/getid3.php' (include_path='.;C:\php\pear') in C:\Apache24\htdocs\podcast\core\episodes.php:211 Stack trace: #0 C:\Apache24\htdocs\podcast\setup\step3.php(49): require() #1 {main} thrown in C:\Apache24\htdocs\podcast\core\episodes.php on line 211

Looking at the INSTALL file says:  "Install Podcast Generator on Ubuntu 22.04"

Does this mean it doesn't work on Windows?

Thanks!

**Comment by charabaruk** [comment, marked as answer]
Unfortunately there are issues with installing 3.0 through 3.2 on Windows using the setup scripts. What has happened is that the path separators in the `absoluteurl` value in your generated config.php were either parsed out as a security precaution or were never saved in the first place. If you open up config.php, find the line that starts with `$absoluteurl = ` (should be line 13) and replace it with the following line:

```php
$absoluteurl = 'C:/Apache24/htdocs/podcast/'; // The location on the server
```

**Reply by JayTea144** [reply]
Thanks!  That fixed the Fatal error.

I'm getting this at the top of every page I navigate:

>Deprecated: Optional parameter $path declared before required parameter $key is implicitly treated as a required parameter in C:\Apache24\htdocs\podcast\core\misc\configsystem.php on line 123

>Deprecated: Optional parameter $path declared before required parameter $content is implicitly treated as a required parameter in C:\Apache24\htdocs\podcast\core\freebox.php on line 20


Also, as I testing it out, I'm unable to successfully upload any file content.

>rush - yyz.mp3 was not uploaded successfully


And it looks like there may be a limit.  Can I adjust the limit somewhere?

>Warning: POST Content-Length of 8674738 bytes exceeds the limit of 8388608 bytes in Unknown on line 0

Thanks,
Jim

**Reply by charabaruk** [reply]
The deprecation warnings and the upload limit are things that need to be configured in php.ini in your PHP installation (and for the latter, possibly also in your Apache config). For the upload limit config in PHP, [this might help](https://stackoverflow.com/a/2184541/5697), but for the warnings and any necessary Apache config, you are unfortunately on your own.

**Reply by JayTea144** [reply]
Thank you so much!

I was able to fix the deprecation warnings by changing the variables in the php.ini file:
`error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT`

And allow bigger posts and uploads from that link you posted.
`post_max_size = xxM`
`upload_max_filesize = yyM`

I didn't have to change anything in Apache config.

Thanks again!
